core/domain/prod_validation_jobs_one_off.py

```diff
@@ -110,116 +288,3206 @@ def validate(cls, item):
+        return [cls._validate_collection_model_version_from_item_id]
+
+
+class CollectionSummaryModelValidator(BaseModelValidator):
+    """Class for validating CollectionSummaryModel."""
+
+    @classmethod
+    def _get_model_id_regex(cls, item):
+        return '.'
+
+    @classmethod
+    def _get_json_properties_schema(cls, item):
+        non_negative_int_schema = {
+            'type': 'int',
+            'validators': [{
+                'id': 'is_at_least',
+                'min_value': 0
+            }]
+        }
+        ratings_schema = {
+            'type': 'dict',
+            'properties': [{
+                'name': '%s' % rating_value,
+                'schema': non_negative_int_schema,
+            } for rating_value in ['1', '2', '3', '4', '5']]
+        }
+
+        if item.ratings and len(item.ratings.keys()):
+            return {'ratings': ratings_schema}
+        else:
+            return {}
+
+    @classmethod
+    def _get_model_domain_object_instances(cls, item):
+        return []
+
+    @classmethod
+    def _get_commit_cmd_domain_class(cls):
+        return None
+
+    @classmethod
+    def _get_external_id_relationships(cls, item):
+        return {
+            'collection_model': (
+                collection_models.CollectionModel, [item.id]),
+            'collection_rights_model': (
+                collection_models.CollectionRightsModel, [item.id]),
+            'owner_user_model': (
+                user_models.UserSettingsModel, item.owner_ids),
+            'editor_user_model': (
+                user_models.UserSettingsModel, item.editor_ids),
+            'viewer_user_model': (
+                user_models.UserSettingsModel, item.viewer_ids),
+            'contributor_user_model': (
+                user_models.UserSettingsModel, item.contributor_ids)
+        }
+
+    @classmethod
+    def _validate_contributors_summary(cls, item):
+        """Validate that contributor ids match the contributor ids obtained
+        from contributors summary.
+
+        Args:
+            item: CollectionSummaryModel to validate.
+        """
+        contributor_ids_from_contributors_summary = (
+            item.contributors_summary.keys())
+        if sorted(item.contributor_ids) != sorted(
+                contributor_ids_from_contributors_summary):
+            cls.errors['contributors summary check'] = (
+                'Model id %s: Contributor ids: %s do not match the contributor '
+                'ids obtained using contributors summary: %s') % (
+                    item.id, (',').join(sorted(item.contributor_ids)),
+                    (',').join(
+                        sorted(contributor_ids_from_contributors_summary)))
+
+    @classmethod
+    def _validate_language_code(cls, item):
+        """Validate that language code is present in allowed language codes.
+
+        Args:
+            item: CollectionSummaryModel to validate.
+        """
+        allowed_language_codes = [
```

Comment:
You can use `utils.is_valid_language_code` here.

Reply:
Done.